src/syscall/syscall_libc.go

```diff
@@ -1,46 +1,66 @@
-// +build darwin
+// +build darwin nintendoswitch
 
 package syscall
 
 import (
+	"errors"
 	"unsafe"
 )
 
+var (
+	notImplemented = errors.New("syscall: not implemented")
+)
+
 func Close(fd int) (err error) {
-	return ENOSYS // TODO
+	errN := libc_close(int32(fd))
```

Comment:
I still wonder why you need this? This PR does not include any other mention of the `Close` function. And while this is a useful addition, it may be better done separately. For example: it should ideally return a correct error.

The same goes for all other libc changes below, I don't see how they are necessary for minimal Nintendo Switch support?